expression/builtin_cast_vec_test.go

```diff
@@ -55,6 +55,7 @@ var vecBuiltinCastCases = map[string][]vecExprBenchCase{
 		{retEvalType: types.ETJson, childrenTypes: []types.EvalType{types.ETJson}},
 		{retEvalType: types.ETJson, childrenTypes: []types.EvalType{types.ETString}, geners: []dataGenerator{&jsonStringGener{}}},
 		{retEvalType: types.ETJson, childrenTypes: []types.EvalType{types.ETDecimal}},
+		{retEvalType: types.ETDatetime, childrenTypes: []types.EvalType{types.ETReal}},
```

Comment:
Please add cases for `mysql.TypeDate`, we can use [`childrenFieldTypes`](https://github.com/pingcap/tidb/blob/2d28a2703f50c99642bdaec7e47901da79187536/expression/bench_test.go#L472) here.

Reply:
Could you provide more details, please?
if I understand correctly, `childrenFieldTypes` is for input datatype only?
How can we change output type?